### PR TITLE
added service hub component and setup project for service hub

### PR DESCRIPTION
**Pull request by heejaechang**
service hub component is basically thin layer that deals with converting data to pass in to RemoteWorkspace

...

commit with change change -
https://github.com/dotnet/roslyn/commit/bbf251e40b9ae08cf256e6813a79d59d27819be1

tracking issue for work left
https://github.com/dotnet/roslyn/issues/12778

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @mattwar @KevinH-MS @dotnet/roslyn-ide @CyrusNajmabadi @panopticoncentral can you take a look?

File: NuGet.Config [hunk 1]
```diff
@@ -24,6 +24,7 @@
     <add key="myget.org symreader" value="https://dotnet.myget.org/F/symreader/api/v3/index.json" />
     <add key="myget.org symreader-portable" value="https://dotnet.myget.org/F/symreader-portable/api/v3/index.json" />
     <add key="myget.org roslyn-master-nightly" value="https://dotnet.myget.org/F/roslyn-master-nightly/api/v3/index.json" />
+    <add key="myget.org devcore" value="https://www.myget.org/F/vs-devcore/api/v3/index.json" />
```
Comment (panopticoncentral) [inline NuGet.Config]: >devcore [](start = 24, length = 7)

Is this used anywhere?

**Comment by panopticoncentral** [comment]
:+1: On the ServiceHub parts, I think we can simplify some but it will be easier to do it in master than in a review. Would still be good to have diag person review the out of proc diagnostics stuff.

**Comment by CyrusNajmabadi** [comment]
I'm wary of this going in.  I'd rather do passes over the PR until it feels good enough to be actually in master.  If we do want this checked in somewhere without full PR signoff, then i'd rather it be in some feature branch.

Master needs to stay clean, highly stable and with very high quality code.  This piece is large enough that really dont' want it in master until we fill it's very close to ship quality.

File: NuGet.Config (hunk 1 above)
Comment (heejaechang) [inline NuGet.Config]: that's where service hub nuget package is coming.

**Comment by heejaechang** [comment]
this PR is merged to this PR.
https://github.com/dotnet/roslyn/pull/12774

**Pull request closed without merging**